### DISCUSSION TITLE
MotoROS2: Fatal Error Alarm 8011 [5]

### DISCUSSION THREAD

**Discussion by shobhitagg** [Yaskawa-Global/motoros2#125] [Troubleshooting] [answered]
I follow all the [installation steps](https://github.com/Yaskawa-Global/motoros2/blob/main/README.md#installation) and steps for [setting up micro-ros-agent](https://github.com/Yaskawa-Global/motoros2/blob/main/README.md#linux-debianubuntu) as per the [README.md](https://github.com/Yaskawa-Global/motoros2/blob/main/README.md) file.

When I start the micro_ros_agent on the workstation, I get the following messages. Seems like the session establishes and then closes immediately after.
![Screenshot 2023-08-15 162052](https://github.com/Yaskawa-Global/motoros2/assets/8982264/9412a69e-f12d-442f-9905-6115078495cb)

The controller shows the following error. Attaching [PANELBOX.LOG](https://github.com/Yaskawa-Global/motoros2/files/12352807/PANELBOX.LOG)

```
MotoROS2: Fatal Error
Alarm code:8011 Subcode:5


Ref: FL6BN
```

**Comment by ted-miller** [comment]
Hi @shobhitagg 

That alarm indicates there was a failure [creating the node](https://github.com/Yaskawa-Global/motoros2/blob/fa34ceeffb2464b44329461ce03b6a9fc04f97a7/src/CommunicationExecutor.c#L176). 

1) What version of ROS2 are you using? (Foxy, Humble, etc) Additionally, did you build the micro_ros_agent for that same version?

2) Are you using the precompiled [release binary](https://github.com/Yaskawa-Global/motoros2/releases/tag/0.1.0) or building MotoROS2 from source?

3) Can you please provide a copy of the [Debug log](https://github.com/Yaskawa-Global/motoros2/blob/main/doc/troubleshooting.md#debug-log-client) output and the [YAML configuration](https://github.com/Yaskawa-Global/motoros2#configuration) file?

4) Have you tried using the [docker image](https://github.com/Yaskawa-Global/motoros2#using-docker-linux-only) for the micro-ros-agent? (Rule out any issues with the build)

**Reply by gavanderhoorn** [reply]
I would indeed suggest to use the micro-ROS Agent Docker image instead of building it from source.

We've not come across a reason to build it ourselves, unless we were fixing/debugging an issue with the Client<->Agent connection.

**Reply by shobhitagg** [reply]
1. Humble
2. Using release binary `motoros2_yrc1000_humble.out`
3. Here are both the files: 
[Debug log](https://github.com/Yaskawa-Global/motoros2/files/12360457/20230816T102244_motoros2_debug_log.txt) (_generated by running micro-ros-agent using docker image_)
[motoros2_config.zip](https://github.com/Yaskawa-Global/motoros2/files/12360492/motoros2_config.zip)

4. Yes, I also tried the docker image. Still has the same error.
![image](https://github.com/Yaskawa-Global/motoros2/assets/8982264/a5c7ce2e-5af9-487a-8d3a-07526286af57)

**Reply by gavanderhoorn** [reply]
Could you please post another debug log which captures the complete startup process?

Perhaps starting the logger before turning on the controller would be easiest.

---

Edit: no longer needed.

**Comment by gavanderhoorn** [comment, marked as answer]
It's likely the problem is the value configured for the `node_namespace`:

```yaml
node_namespace: "yk-creator"
```

Dashes (ie: `-`) are not legal in ROS resource names.

The debug log also shows:

```
[2023-08-16 10:22:50.315183] [192.168.1.73:55479]: rclc_node_init_with_options = 202
```

The `202` maps to [`RCL_RET_NODE_INVALID_NAMESPACE`](https://github.com/ros2/rcl/blob/cc9ea2810ff7c181ccdd43dc1f5c336f531fa8d8/rcl/include/rcl/types.h#L59-L60).

I'd suggest removing the `-` (or replacing it with a `_`).

**Reply by gavanderhoorn** [reply]
@ted-miller: we might want to consider reporting some of these RCL(C) initialisation errors, or defining some subcodes for them.

Especially the `NODE`, `PUBLISHER`, `SUBSCRIPTION`, etc categories seem to be caused by things which could often be solved by users without needing to post on the tracker here.
